Lib/test/test_io.py

```diff
@@ -604,10 +604,15 @@ def test_large_file_ops(self):
             support.requires(
                 'largefile',
                 'test requires %s bytes and a long time to run' % self.LARGE)
-        with self.open(support.TESTFN, "w+b", 0) as f:
-            self.large_file_ops(f)
-        with self.open(support.TESTFN, "w+b") as f:
-            self.large_file_ops(f)
+        try:
+            with self.open(support.TESTFN, "w+b", 0) as f:
+                self.large_file_ops(f)
+            with self.open(support.TESTFN, "w+b") as f:
+                self.large_file_ops(f)
+        except OSError as e:
+            if e.errno == errno.ENOSPC:
+                self.skipTest("No space left on device")
+            raise
```

Comment:
Do this raise stop the test? Or it just print in the stdout!?

Reply:
`raise` will re-raise the current exception, effectively stopping the single test.

I added a line `raise RuntimeException` in the test body to test this behavior, this is the result:

```
======================================================================
ERROR: test_large_file_ops (test.test_io.CIOTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gcappellotto/src/cpython/Lib/test/test_io.py", line 608, in test_large_file_ops
    raise RuntimeError
RuntimeError

======================================================================
ERROR: test_large_file_ops (test.test_io.PyIOTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gcappellotto/src/cpython/Lib/test/test_io.py", line 608, in test_large_file_ops
    raise RuntimeError
RuntimeError

----------------------------------------------------------------------

Ran 606 tests in 38.472s

FAILED (errors=2, skipped=8)
test test_io failed
test_io failed in 38 sec 661 ms

== Tests result: FAILURE ==

1 test failed:
    test_io

Total duration: 38 sec 714 ms
Tests result: FAILURE
```